### DISCUSSION TITLE
`export-env` with `export use`

### DISCUSSION THREAD

**Discussion by Zoybean** [nushell/nushell#11812] [Q&A] [answered]
This is maybe a bit of a hack, but I've got a few modules defining commands that I use frequently, and one of those modules cares about (and sets) a few specific environment variables. My config.nu `use`s a `mod.nu` that re-exports all the other modules, but it seems like it doesn't re-export the environment from those modules.

Part of the issue was #7291, but even after merging my `export-env` blocks, I still need to explicitly `use foo` in order to get its environment definitions.
All the command definitions within the modules appear to be available as expected, it's just the environment that's not working.

Am I misunderstanding how these commands are meant to work, or is this a bug in nushell?

---

`$nu.config-path`:
```
use ~/bin *
```
`~/bin/mod.nu`:
```
export use foo.nu
```
`~/bin/foo.nu`:
```
export-env { $env.FOO = "bar" }
def cmd [] { "working" }
```

shell:
```
> $env.FOO
Error: nu::shell::column_not_found

  × Cannot find column
   ╭─[entry #1:1:1]
 1 │ $env.FOO
   · ──┬─ ─┬─
   ·   │   ╰── cannot find column 'FOO'
   ·   ╰── value originates here
   ╰────
> foo cmd
working
> use foo
> $env.FOO
bar
```

**Comment by fdncred** [comment]
I'm not sure. Maybe @kubouch has an idea?

**Comment by kubouch** [comment]
I think you'd need to wrap the `export use foo.nu` into an `export-env {}` block inside `~/bin/mod.nu` because only `export-env` blocks are evaluated at runtime. Maybe `export-env { source-env foo.nu }` would convey the intent the most clearly.

**Reply by Zoybean** [reply]
Wrapping `use` or `export use` in `export-env` did it as far as the environment changes, but now the other members of the module are unloaded.
For now, I'll just duplicate my `use`s like this:
```nu
export-env {
  # load the environment
  use foo.nu 
}
# load the definitions
export use foo.nu
```

I couldn't get it to work with `source-env` either - it complained that it couldn't find `foo.nu` from `mod.nu`:
```
Error: nu::shell::file_not_found

  × File not found
   ╭─[C:\Users\...\bin\mod.nu:2:14]
 1 │ export-env {
 2 │   source-env foo.nu
   ·              ───┬──
   ·                 ╰── file not found
 3 │ }
   ╰────
```

**Reply by kubouch** [reply]
Yes, the definitions need to be loaded manually separately. The `source-env` not working seems like a bug, could you file an issue with it?

**Reply by Zoybean** [reply]
Raised #12042

**Comment by ghost** [comment, marked as answer]
Just gonna drop a link here: https://github.com/nushell/nushell/issues/6973

it was "intended" back in 2022
